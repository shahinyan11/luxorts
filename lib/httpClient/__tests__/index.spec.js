import Axios from 'axios';
import qs from 'qs';

import buildHttpClient from '../index';

jest.mock('axios', () => ({
  create: jest.fn(),
}));

jest.mock('qs', () => ({
  stringify: jest.fn(),
}));

describe('buildHttpClient()', () => {
  it('returns new instance of axios', () => {
    buildHttpClient();

    const createArgs = Axios.create.mock.calls[0][0];
    expect(createArgs).toMatchSnapshot();

    createArgs.paramsSerializer({});
    expect(qs.stringify).toHaveBeenCalledWith({}, { arrayFormat: 'brackets' });
  });
});
