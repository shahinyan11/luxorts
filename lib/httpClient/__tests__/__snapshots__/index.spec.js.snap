// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildHttpClient() returns new instance of axios 1`] = `
Object {
  "baseURL": "http://localhost:3000/api/v1",
  "headers": Object {
    "common": Object {},
  },
  "paramsSerializer": [Function],
}
`;
